### PR TITLE
Remove parallelization in batch sample.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Removes parallel calls into python simulators when sampling. Also fixes datatype outputs to match that of that high speed C++ ops (int8).

### CODE REVIEW

File: tensorflow_quantum/core/ops/batch_util.py [hunk 1]
```diff
@@ -538,30 +412,18 @@ def batch_sample(circuits, param_resolvers, n_samples, simulator):
...
+        if len(c.all_qubits()) == 0:
+            continue
 
-        pool.starmap(_sample_worker_func, input_args)
+        qb_keys = [(q, str(i)) for i, q in enumerate(sorted(c.all_qubits()))]
+        c_m = c + cirq.Circuit(cirq.measure(q, key=i) for q, i in qb_keys)
+        run_c = cirq.resolve_parameters(c_m, resolver)
+        bits = simulator.sample(run_c, repetitions=n_samples)
+        flat_m = bits[[x[1] for x in qb_keys]].to_numpy().astype(np.int8)
+        return_array[batch, :, :len(qb_keys)] = flat_m[:, ::-1]
```
Comment (zaqqwerty) [inline tensorflow_quantum/core/ops/batch_util.py:426]: Would it work to do one fewer cycles of reversing: `return_array[batch, :, :len(qb_keys)] = flat_m[:, ::-1]` -> `return_array[batch, :, biggest_circuit - len(qb_keys):] = flat_m`, then just `return return_array`?

**Review by zaqqwerty — CHANGES_REQUESTED** [review]
Thanks for getting this over the finish line for me, one question

File: tensorflow_quantum/core/ops/batch_util.py (hunk 1 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/batch_util.py]: Oops, good call. Done.

**Review by zaqqwerty — APPROVED** [review]
LGTM

**Pull request merged**